metric/example_test.go

```diff
@@ -142,6 +142,35 @@ func ExampleMeter_upDownCounter() {
 	}
 }
 
+// Gauges can be used to record non-additive values when changes occur.
+//
+// Here's how you might report the current speed of a cpu fan.
+func ExampleMeter_gauge() {
+	speedGauge, err := meter.Int64Gauge(
+		"cpu.fan.speed",
+		metric.WithDescription("Speed of CPU fan"),
+		metric.WithUnit("RPM"),
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	getCPUFanSpeed := func() int64 {
+		// hard-code 1500 RPM for demonstrative purposes
+		// do the real work to get the cpu fan speed
+		return 1500
+	}
+
+	_ = func() {
+		go func() {
+			for {
+				time.Sleep(time.Second * 3)
+				speedGauge.Record(context.Background(), getCPUFanSpeed())
+			}
+		}()
+	}
```

Comment:
The point of the synchronous gauge is that the measurement cycle is decoupled from the recording process. Instead of having the recording loop handle timing, we can have this be a more explicit subscription. E.g.

```suggestion
	fanSpeedSubscription := make(chan int64, 1)
	go func() {
		// Use fixed speeds for demonstration. In the real world applications,
		// these would be the actual fan speed measurements.
		speeds := []int64{1500, 1250, 850, 900}
		for _, speed := range speeds {
			// Synchronous gauges are used when the measurement cycle is
			// synchronous to an external change. Simulate that external
			// cycle here.
			time.Sleep(time.Duration(rand.Intn(3)) * time.Second)
			fanSpeedSubscription <- speed
		}
		close(fanSpeedSubscription)
	}()

	ctx := context.Background()
	for speed := range fanSpeedSubscription {
		speedGauge.Record(ctx, speed)
	}
```

Reply:
Thank you for the detailed explain. I updated the code to decouple measurement cycle from recoding process.